docs: consolidate identity schema docs

tomekpapiernik opened the pull request:
<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

This text will be included in the changelog. If applicable, include links to documentation or pieces of code.
If your change includes breaking changes please add a codeblock documenting the breaking change:

```
BREAKING CHANGES: This patch changes the behavior of configuration item `foo` to do bar. To keep the existing
behavior please do baz.
```
-->

This PR consolidates the existing Identity / Identity schema documentation. Some new content has been added (such as the "Overview" document) while old docs were updated and tweaked.

## Related Issue or Design Document

<!--
If this pull request

1. is a fix for a known bug, link the issue where the bug was reported in the format of `#1234`;
2. is a fix for a previously unknown bug, explain the bug and how to reproduce it in this pull request;
2. implements a new feature, link the issue containing the design document in the format of `#1234`;
3. improves the documentation, no issue reference is required.

Pull requests introducing new features, which do not have a design document linked are more likely to be rejected and take on average 2-8 weeks longer to
get merged.

You can discuss changes with maintainers either in the Github Discusssions in this repository or
join the [Ory Chat](https://www.ory.sh/chat).
-->

https://github.com/ory-corp/cloud/issues/2654

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability.
      If this pull request addresses a security vulnerability,
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation within the code base (if appropriate).

## Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution
you did and what alternatives you considered, etc...
-->

vinckr commented on docs/guides/manage-identities/01_overview.mdx:
```diff
@@ -0,0 +1,65 @@
+on a website or an application that uses Ory, for example online store customers, file sharing service users, or company
+contractors signing up to use internal systems.
+
+The Identity isn't limited to humans. Ory can create Identities for programmatic entities such as IoT devices, client
```
```suggestion
The identity isn't limited to humans. Ory can create identities for programmatic entities such as IoT devices, client
```

IMO it looks more natural to have it lowercase. I get that you want to use "Identity" as a proper term in the Ory context, but IMO its still clear if lowercase and looks less weird.
(at least when its not "Ory Identities")

tomekpapiernik replied:
Thanks! I think you're right. I decided to do `identity` and `Identity Schema`.

vinckr commented on src/sidebar.js:
```diff
@@ -84,11 +84,10 @@ module.exports = {
           type: "category",
           label: "Managing Identities",
```
```suggestion
          label: "Managing identities",
```

tomekpapiernik replied:
Let's do them all in one go, not in this PR ;)

vinckr commented on docs/guides/manage-identities/20_create-identities.mdx:
```diff
@@ -1,22 +1,10 @@
-id: managing-users-identities
-title: Create and invite identities
+id: create-identities
+title: Create Identities
```
```suggestion
title: Create identities
```

To be consistent I would lowercase this. see above and below, we sometimes have it lowercased, sometimes uppercased as "proper term". 
But I think it would be less errorprone to lowercase everything that is not a super Ory specific term like Identity Schema.

vinckr commented on docs/guides/manage-identities/15_customize-identity-schema.mdx:
```diff
@@ -1,40 +1,22 @@
 ---
 id: customize-identity-schema
-title: Customize identity schema
+title: Customize Identity schema
```
```suggestion
title: Customize identity schema
```
or
```suggestion
title: Customize Identity Schema
```
if we want to establish Identity Schema as proper term.

vinckr commented on docs/guides/manage-identities/05_identity-schema.mdx:
````diff
@@ -0,0 +1,293 @@
+}
+```
+
+### Identifier for Username and Password flows
````
```suggestion
### Identifier for username and password flows
```

vinckr commented on docs/guides/manage-identities/05_identity-schema.mdx:
````diff
@@ -0,0 +1,293 @@
+}
+```
+
+### Email & password
````
```suggestion
### Email and password
```

should we ban ampersand, I think that would make it easier to handle, wdyt? we have a wild mix atm :D

vinckr commented on docs/guides/manage-identities/05_identity-schema.mdx:
```diff
@@ -0,0 +1,293 @@
+[Username and Password Credentials](../../kratos/concepts/credentials/username-email-password) contains more information and
+examples about credentials usage.
+
+Note that `format` field of the Identity schema will perform validation of the given trait. In this example, the email address is
```
```suggestion
Note that the `format` field of the Identity schema will perform validation of the given trait. In this example, the email address is
```

vinckr commented on docs/guides/manage-identities/15_customize-identity-schema.mdx:
```diff
@@ -1,40 +1,22 @@
+This document describes how to create custom Identity schemas.
 
-This guide describes the steps needed to create custom Identity Schemas.
+## Create custom Identity schema
```
```suggestion
## Create custom identity schema
```
or
```suggestion
## Create custom Identity Schema
```
see above

vinckr commented on docs/guides/manage-identities/30_account-recovery.mdx:
```diff
@@ -3,12 +3,12 @@ id: account-recovery
 ---
 
-It is possible to initiate account recovery for identities using the Admin API endpoints.
+You can initiate account recovery for Identities using the Admin API endpoints.
```
there is a few typos in this doc but we can fix them after the PR,

and a link to https://console.ory.sh/projects/current/session that leads nowhere, needs to be updated to https://console.ory.sh/projects/current/session-settings.

also the selfhosted kratos example doesnt make sense. it would be enough to use the cURL example with selfhosted no?

tomekpapiernik replied:
I looked for typos and fixes some wonky sentences. For self-hosted it's okay to either edit the config or call the API, idk which one is better.

vinckr commented on docs/guides/manage-identities/35_best-practices.mdx:
```diff
@@ -0,0 +1,63 @@
+:::
+
+The Identity Schema isn't the right place to store data that should be obfuscated from the user. Users can see traits and other
+data - except credentials - using the `sessions/whoami` endpoint. Users are also able to edit identity traits. Visit
+[this issue](https://github.com/ory/kratos/issues/47) for information on a feature that can be configured to prevent the end-user
+from viewing/editing identity traits.
```
```suggestion
data - except credentials - using the `sessions/whoami` endpoint. Users are also able to edit identity traits. Visit
[this document](https://www.ory.sh/docs/guides/manage-identities/managing-users-identities-metadata) for information how to use
metadata that can not be viewed or changed by the end-user.
```

vinckr commented on docs/kratos/concepts/credentials/username-email-password.mdx:
````diff
@@ -77,9 +77,6 @@ identity:
       # url: base64://...
 ```
 
-If you don't know what that means, please read the [Identity Schema Chapter](../../concepts/identity-schema.mdx) in the docs'
````
should we not link the new identity schema doc here?

tomekpapiernik replied:
Added this back

aeneasr commented on vercel.json:
```diff
@@ -633,6 +633,26 @@
+    },
+    {
+      "source": "docs/guides/manage-identities/managing-users-identities-metadata",
+      "destination": "https://www.ory.sh/docs/guides/manage-identities/identity-metadata-traits",
```
For SEO it is usually a good idea to have "user", "identity" and "management" in the URL. This will rank it higher when people search for "user/identity management open source" for example. I don't think many people look for "identity traits open source"

tomekpapiernik replied:
I'm not SEOing document IDs, I'm trying to come up with filenames+IDs that accurately represent the content. 
Please, let's keep SEO out of this :D

aeneasr replied:
Good SEO is paramount for documentation, it helps drive new users to Ory, makes pages easy to find on Google (which is what most users use to find content if they are facing an issue). Keeping good SEO practices in documentation is non-negotiable.

aeneasr replied:
Here are some resources to educate yourself on SEO and documentation and tech writing:

* https://docs.readthedocs.io/en/stable/guides/technical-docs-seo-guide.html
* https://idratherbewriting.com/2013/09/22/how-to-search-engine-optimize-your-help-content-or-documentation/ (very dated but premise remains the same)

aeneasr commented on vercel.json:
```diff
@@ -633,6 +633,26 @@
+    },
+    {
+      "source": "docs/guides/manage-identities/managing-users-identities",
+      "destination": "https://www.ory.sh/docs/guides/manage-identities/create-identities",
```
Same here as in the SEO comment below. What would be a good google search term that we would want to rank this site for prominently?

tomekpapiernik replied:
See the comment above, please

tomekpapiernik commented on docs/guides/manage-identities/01_overview.mdx:
```diff
@@ -0,0 +1,65 @@
+sidebar_label: Overview
+---
+
+Ory Identity Service (Kratos) uses identities to identify entities that sign up or authenticate. These can be humans that sign up
```
Too much "Ory'ing" IMO. We're in Ory docs, it's the Ory Identity Service... ;)

aeneasr replied:
The product name is "Ory Kratos", so it has to be named like that if the name is included, even if it is repetitive.

tomekpapiernik commented on docs/guides/manage-identities/05_identity-schema.mdx:
````diff
@@ -0,0 +1,293 @@
+}
+```
+
+### Demo profile
+
+:::warning
+
+This preset is experimental. Use it for PoCs and demos only.
+
+:::
````
Let's, but only when the UI uses this name as well.

tomekpapiernik replied:
The PR is here: https://github.com/ory-corp/cloud/pull/2831/

Updated the doc 👍🏻

tomekpapiernik commented on docs/guides/manage-identities/05_identity-schema.mdx:
```diff
@@ -0,0 +1,293 @@
+
+### Identifier for username and password flows
+
+You can configure Ory Identity Service (Kratos) to use specific fields as the identity's identifier. In this example, the
```
Too much "Ory'ing" IMO. We're in Ory docs, it's the Ory Identity Service... ;)

aeneasr replied:
See above, please follow my lead on this.

tomekpapiernik replied:
Okay

vinckr commented on docs/guides/manage-identities/15_customize-identity-schema.mdx:
```diff
@@ -168,9 +149,9 @@ you want to know more about these types, please refer to the
+![Example Identity Schema with GitHub and Phone Number Fields](../_static/manage-identities/2.png)
 
-### Change existing Identity Schemas
+### Change existing Identity schemas
```
```suggestion
### Change existing Identity Schemas
```

So lets establish "Identity Schema" as proper ory specific term?

tomekpapiernik replied:
Sorry, missed that.

vinckr reviewed (commented):
Dont have much anything to add to aeneas review, LGTM

tomekpapiernik commented on docs/guides/manage-identities/20_create-identities.mdx:
```diff
@@ -47,7 +35,7 @@ curl --request POST -sL \\
   <TabItem value="kratos">
     <SelfHostedKratos/>
-    <p>And use it to create an identity:</p>
+    <p>And use it to create an Identity:</p>
```
```suggestion
    <p>And use it to create an identity:</p>
```

tomekpapiernik commented on docs/guides/manage-identities/30_account-recovery.mdx:
```diff
@@ -47,27 +47,27 @@ curl --request POST -sL \
-	"github.com/ory/client-go/client"
-	"github.com/ory/client-go/client/admin"
-	"github.com/ory/client-go/models"
+  "fmt"
```
Yes, by Prettier.

tomekpapiernik commented on vercel.json:
```diff
@@ -633,6 +633,26 @@
       "permanent": false
+    },
+    {
+      "source": "docs/kratos/concepts/identity-schema",
```
Removed

tomekpapiernik commented:
@aeneasr As discussed yesterday: 

- Added more descriptive, search engine-friendly titles to all documents. 
- Updated document IDs to be more SEO-friendly
- Updated filenames to mirror document IDs

Decided to stick with linking using document ID rather than switch over to linking using filenames. 
This way:
- Links between documents will be a tiny bit shorter (no `mdx` and `05` to type)
- Links between documents will still work even if the document order gets adjusted through modifying filenames. 
- We still will be able to send users to specific headings in documents. It's not something we use extremely often, but it's a nice thing to have in the tool belt. It's not possible to achieve when linking using filenames AFAIK.

aeneasr commented:
Great job, thank you!
